### ISSUE TITLE
Need to improve the code style & formatting regulation

### ISSUE THREAD

**Issue by jiajunwang** [apache/helix#349]
The current format file helix-style.xml has very limited configured items. This leaves too much flexibility to Helix developers. We shall add more items to it.

The known missing ones:
- Wildcard Imports
- Annotation wrapping
Could be more.

Please comment on this issue and we will modify the format file accordingly.

**Comment by narendly** [comment]
+1

If possible, let us also add the Apache license at the top of every file.

Hunter

**Comment by pkuwm** [comment]
How about the order of imports, to make the imports consistent in the project? Like:
:::java
import java.*
import javax.*

import com.*

import net.*

import org.*

import static *

**Comment by i3wangyi** [comment]
You know there's a tool to auto check every commit see if it meets the requirement? https://github.com/checkstyle/checkstyle

The detailed work of styling issue can be breakdown into
1. A better java coding style.xml
2. The integration of the tool that auto checks, stops bad commit before checking in.

**Comment by pkuwm** [comment]
I will take this. And let me add more items here to discuss about.

**Comment by jiajunwang** [comment]
Wonderful! Many thanks.

**Comment by dasahcc** [comment]
Just a reminder, there could be a bug in style file as well. When we format @Override annotation, it moved to same line as function declaration. Let's fix it.

**Comment by pkuwm** [comment]
Sure, will do. Thanks, Junkai.

**Comment by pkuwm** [comment]
How about we follow some rules of google java style: https://google.github.io/styleguide/javaguide.html?
Like:
- change Column limit to 100

I will submit a PR to address some styles like imports order.

**Comment by pkuwm** [comment]
Column limit is already 100 in helix-style.xml. Do we consider just using an IntelliJ style file and importing the file, instead of using eclipse code format plugin ?

**Comment by narendly** [comment]
Could we keep the change minimal?

**Comment by pkuwm** [comment]
1. It seems avoiding wildcard imports cannot be set in the config file with eclipse code formatter plugin. It can be configured in Intellij.
2. Override annotation wrap is already formatted in the current style, like below:
```
  @Override public boolean removeListener
```
Could be formatted:
```
  @Override
  public boolean removeListener
```
We'll find out any case I miss.

3. Due to [Sun's convention](https://checkstyle.org/styleguides/sun-code-conventions-19990420/CodeConventions.doc2.html#a277), Apache License copyright comment should be added from the first line, before `package` statement. We are now having package statement at first line, which I think should be changed.

In summary, current style file seems good enough. For `imports` setting and Apache License, we might need to manually change the settings in Intellij -- how about creating a coding style wiki?
